src/internal/collection/postgres_collection.go

```diff
@@ -188,8 +195,20 @@ type postgresReadOnlyCollection struct {
 func (c *postgresCollection) get(ctx context.Context, key string, q sqlx.QueryerContext) (*model, error) {
 	result := &model{}
 	queryString := fmt.Sprintf("select proto, updatedat from collections.%s where key = $1;", c.table)
-	if err := sqlx.GetContext(ctx, q, result, queryString, key); err != nil {
-		return nil, c.mapSQLError(err, key)
+	// wrap sqlx.GetContext() in a retry to mitigate database connection flakiness.
+	if err := backoff.RetryUntilCancel(ctx, func() error {
+		if err := sqlx.GetContext(ctx, q, result, queryString, key); err != nil {
+			return c.mapSQLError(err, key)
+		}
+		return nil
+	}, backoff.RetryEvery(time.Second), func(err error, d time.Duration) error {
+		if strings.Contains(err.Error(), "broken pipe") || strings.Contains(err.Error(), "unexpected EOF") {
```

Comment:
Let's extract this into an errutils.IsDatabaseDisconnect because I'm sure there's some third message we'll want to test for eventually.  We should also check to see if `errors.Is(err, io.ErrUnexpectedEOF)` catches the error we care about; just to make this condition a little more precise.